File: swift/word-count/Sources/WordCount/WordCount.swift
struct WordCount {
    let words: String

    func count() -> [String: Int] {
        words.split { !$0.isLetter && !$0.isNumber }
                .reduce(into: [:]) { acc, w in
                    acc[String(w).lowercased(), default: 0] += 1
                }
    }
}

File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    sentence
    |> String.downcase()
    |> String.split(~r/[^[:alnum:]-']/u, trim: true)
    |> Enum.reduce(%{}, fn s, acc ->
      acc |> Map.update(s |> String.trim("'"), 1, &(&1 + 1))
    end)
  end
end
